File: socialuni-sdk/src/main/java/com/socialuni/sdk/model/RO/user/SocialuniUserRO.java
package com.socialuni.sdk.model.RO.user;

import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * @author qinkaiyuan 查询结果可以没有set和空构造，前台传值可以没有get
 * @date 2019-08-13 23:34
 */
@Data
@NoArgsConstructor
public class SocialuniUserRO {
    private String id;
    private String nickname;
    private String avatar;
    private String gender;

    public SocialuniUserRO(SocialuniUserRO userRO) {
        this.id = userRO.getId();
        this.nickname = userRO.getNickname();
        this.avatar = userRO.getAvatar();
        this.gender = userRO.getGender();
    }
}

File: socialuni-js-packages/packages/socialuni-user/src/model/SocialuniUserRO.ts
export default class SocialuniUserRO {
    userId: string
    nickname?: string
    avatar?: string
    age?: number
    gender?: number
    birth?: string

    constructor(user: SocialuniUserRO) {
        this.userId = user.userId
        this.nickname = user.nickname;
        this.avatar = user.avatar;
        this.gender = user.gender;
        this.age = user.age;
        this.birth = user.birth;
    }
}
